langlib/lang.regexp/src/main/ballerina/regexp.bal

```diff
@@ -51,8 +51,40 @@ type GroupsAsSpanArrayType SpanAsTupleType[];
 
 type GroupsArrayType GroupsAsSpanArrayType[];
 
+# Splits a string into substrings separated by matches of a regular expression.
+# This finds the the non-overlapping matches of a regular expression and
+# returns a list of substrings of `str` that occur before the first match,
+# between matches, or after the last match.  If there are no matches, then
+# `[str]` will be returned.
+#
+# ```ballerina
+# regexp:RegExp r = re `,`;
+#
+# r.split("Not Valid") ⇒ ["Not Valid"]
+#
+# r.split("abc,cde,efg") ⇒ ["abc", "cde", "efg"]
```

Comment:
```suggestion
# regexp:RegExp r = re `,`;
# r.split("Not Valid") ⇒ ["Not Valid"]
# r.split("abc,cde,efg") ⇒ ["abc", "cde", "efg"]
```

Reply:
Let's not do this.

Reply:
[d5d0ee3](https://github.com/ballerina-platform/ballerina-lang/pull/39096/commits/d5d0ee3fe3895701609512f034d3e641a7f34b3c)